evm/src/executor/backend/mod.rs

```diff
@@ -838,26 +836,11 @@ impl DatabaseExt for Backend {
         }
     }
 
-    fn create_fork(
-        &mut self,
-        fork: CreateFork,
-        journaled_state: &JournaledState,
-    ) -> eyre::Result<LocalForkId> {
+    fn create_fork(&mut self, fork: CreateFork) -> eyre::Result<LocalForkId> {
         trace!("create fork");
         let (fork_id, fork, _) = self.forks.create_fork(fork)?;
         let fork_db = ForkDB::new(fork);
 
-        // there might be the case where a fork was previously created and selected during `setUp`
-        // not necessarily with the current caller in which case we need to ensure that the init
-        // state also includes the caller
-        if let Some(caller) = self.caller_address() {
-            if !self.fork_init_journaled_state.state.contains_key(&caller) {
-                if let Some(account) = journaled_state.state.get(&caller).cloned() {
-                    self.fork_init_journaled_state.state.insert(caller, account);
-                }
-            }
-        }
-
```

Comment:
This was removed since it was causing trouble with https://github.com/foundry-rs/foundry/pull/2541/commits/3d09814c3ef478043793e21e40a2bf5e7b0ade88 .We assume that the caller is not part of the state when **first** selecting the fork. Caller should not be persistent, and this goes against that.


This was first introduced to tackle an error when reverting: https://github.com/foundry-rs/foundry/pull/2989.  However, by removing this block of code the associated test still passes.